clib/ivar.c

```diff
@@ -85,19 +88,32 @@ void ivar_put_with_size( struct ivar iv, void *d, int size )
     log_3("ivar_put_with_size %p %p %d - leave\n", &iv, d, size);
 }
 
-void ivar_put_array( struct ivar iv, struct array *d )
+void ivar_put_array( struct ivar iv, struct array *d, void* vcf )
+{
+    struct ivar_internals *ivi = iv.internals;
+    log_2("ivar_put_array %p %p - enter\n", &iv, d);
+    pthread_mutex_lock( &(ivi->mutex) );
+    array_copy_t* cf = (array_copy_t*) vcf;
+    ivi->data = cf( NULL, d );
+    ivi->full = 1;
+    pthread_cond_broadcast( &(ivi->cond) );
+    pthread_mutex_unlock( &(ivi->mutex) );
+    log_2("ivar_put_array %p %p - leave\n", &iv, d);
+}
+
+void ivar_put_array_shallow( struct ivar iv, struct array *d, int32_t size )
 {
     struct ivar_internals *ivi = iv.internals;
     log_2("ivar_put_array %p %p - enter\n", &iv, d);
     pthread_mutex_lock( &(ivi->mutex) );
-    ivi->data = initArray( NULL, d->elemSize, d->length );
-    copyArray( ivi->data, d );
+    ivi->data = initCopyArray( NULL, size, d );
     ivi->full = 1;
     pthread_cond_broadcast( &(ivi->cond) );
     pthread_mutex_unlock( &(ivi->mutex) );
     log_2("ivar_put_array %p %p - leave\n", &iv, d);
 }
 
+
```

Comment:
Only a single blank line between the other functions concerning ivars.

Reply:
Done